src/main/java/org/calrissian/flowmix/api/aggregator/AbstractGrouplessAggregator.java

```diff
@@ -0,0 +1,140 @@
+ */
+package org.calrissian.flowmix.api.aggregator;
+
+import static java.lang.System.currentTimeMillis;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import static java.util.UUID.randomUUID;
+import org.apache.commons.lang.StringUtils;
+import org.calrissian.flowmix.api.Aggregator;
+import org.calrissian.flowmix.core.model.event.AggregatedEvent;
+import org.calrissian.flowmix.core.support.window.WindowItem;
+import org.calrissian.mango.domain.Tuple;
+import org.calrissian.mango.domain.event.BaseEvent;
+import org.calrissian.mango.domain.event.Event;
+
+/**
+ *
+ * This abstract Aggregator doesn't mantain the whole group, you need to add and
+ * evict the tuples with simple operations and emit them with complex ones.
+ *
+ * @author Miguel A. Fuentes Buchholtz
+ * @param <T> Aggregation result type
+ * @param <F> Field type
+ */
+public abstract class AbstractGrouplessAggregator<T, F> implements Aggregator {
+
+    /**
+     * field to operate with
+     */
+    public static final String OPERATED_FIELD = "operatedField";
+
+    /**
+     * output field
+     */
+    public static final String OUTPUT_FIELD = "outputField";
+
+    /**
+     * fields to group by
+     */
+    protected String[] groupByFields;
+
+    /**
+     * operated field name
+     */
+    protected String operatedField;
+
+    /**
+     * output field set by implementation
+     */
+    protected String outputField = getOutputField();
+
+    /**
+     *
+     * @return output field implementation
+     */
+    protected abstract String getOutputField();
+
+    /**
+     *
+     * @param configuration
+     */
+    @Override
+    public void configure(Map<String, String> configuration) {
+        if (configuration.get(GROUP_BY) != null) {
+            groupByFields = StringUtils.splitPreserveAllTokens(configuration.get(GROUP_BY), GROUP_BY_DELIM);
+        }
+        if (configuration.get(OUTPUT_FIELD) != null) {
+            outputField = configuration.get(OUTPUT_FIELD);
+        }
+        if (configuration.get(OPERATED_FIELD) != null) {
+            operatedField = configuration.get(OPERATED_FIELD);
+        } else {
+            throw new RuntimeException("Aggregator needs a field to operate it. Property missing: " + OPERATED_FIELD);
+        }
+    }
+
+    /**
+     *
+     * @param fieldValue field value to work with after item is added to grouped
+     * values
+     */
+    public abstract void postAddition(F fieldValue);
+
+    /**
+     *
+     * @param item
+     */
+    @Override
+    public void added(WindowItem item) {
+        if (item.getEvent().get(operatedField) != null) {
+            postAddition(((F) item.getEvent().get(operatedField).getValue()));
+        }
+    }
+
+    /**
+     *
+     * @param item item to evict
+     */
+    @Override
+    public void evicted(WindowItem item) {
+        if (item.getEvent().get(operatedField) != null) {
+            evict((F) item.getEvent().get(operatedField).getValue());
```

Comment:
you mean check the generic type F? I don't think that's possible in Java because of the Type Erasure, the other path is using refl*ugh*ection, for example this guy approach: http://www.artima.com/weblogs/viewpost.jsp?thread=208860, but this is too dirty and slow